### ISSUE TITLE
VaadinSecurity.getAuthentication() returns null after successful login

### ISSUE THREAD

**Issue by mikrobestie** [peholmst/vaadin4spring#160]
Hi, 

at first I need to say that I am glad to see that you are working on vaadin spring security module. We are using vaadin4spring in one of our company applications and now I try to integrate new security features to it. 

I used almost the same security configuration as the one in the security sample (SecurityConfiguration.java) and logging the users in is working flawlessly - the UI is loaded, but after the request is completed, it seems that the SecurityContext is cleared and after that, in next request (switching view or polling) when I call:

**vaadinSecurity.getAuthentication()** 

(from @Autowired VaadinSecurity)

I always get null. Is this known issue or I am doing anything wrong? The only difference between our SecurityConfiguration and the one in the sample is that we are using custom AuthenticationProvider. In the security sample, there is only one view.

**Comment by GJRTimmer** [comment]
What you encounter is something I forget to fix within the VaadinSecurity object. We completly switched to the full ```springSecurityFilterChain``` and the filter chain clears the context after it completes. This is by design.

To access the ```Authentication``` object you have to retrieve it from the same place the filter chain has stored it. 

I will try patching the ```VaadinSecurity``` object when I can find a moment the next few days.

In the mean time; check out if this works for you.

```java

@Autowired
private HttpService http;

HttpSession session = http.getCurrentRequest().getSession();

Authentication authentication = ((org.springframework.security.core.context.SecurityContextImpl)session().getAttribute("SPRING_SECURITY_CONTEXT")).getAuthentication();
```

You probably have to play with this a little, I'm doing it from memory.

In short: get the ```Authentication``` object from the HttpSession where it is stored.

**Comment by mikrobestie** [comment]
Thanks, that worked! Is there any chance that it will be fixed in 0.0.4 version?

**Comment by mikrobestie** [comment]
There is a possible flaw, that you can set the session key other than "SPRING_SECURITY_CONTEXT" via HttpSessionSecurityContextRepository.setSpringSecurityContextKey() and thus this method will not work.

**Comment by peholmst** [comment]
I won't release 0.0.4 until this is fixed.

**Comment by GJRTimmer** [comment]
@mikrobestie I'm actually working on it right now. To answer your previous question, yes changing could lead to a problem that's why we already implemented a method to also set the context key on the ```VaadinSecurity``` object, because the ```VaadinSecurity``` has to set the same key the ```HttpSessionSecurityContextRepository``` checks.

```java

@See VaadinSecurity
void setSpringSecurityContextKey(String springSecurityContextKey);

```

**Comment by GJRTimmer** [comment]
@mikrobestie I've pushed the dev/security branch to @peholmst could you checkout this branch build and test it and see if this permanently fixes your issue ?

**Comment by GJRTimmer** [comment]
Will wait with PR until @mikrobestie has confirmed that fix provided within branch dev/security works. I've tried to recreate it but, I still have an unfinished chain so the SecurityContext is still available within my setups.

**Comment by mikrobestie** [comment]
Tried it right now and it works as expected. Thanks for quick fix!

**Comment by GJRTimmer** [comment]
@mikrobestie Thank you for the quick testing. Will submit PR right now.
